Makefile

```diff
@@ -137,7 +137,11 @@ LDFLAGS += $(PROFILE)
 # enable optimizations. slow to build
 ifdef RELEASE
   ifeq ($(NATIVE), osx)
-    CXXFLAGS += -O3
+    ifeq ($(shell $(CXX) -E -Os - < /dev/null > /dev/null 2>&1 && echo fos),fos)
```

Comment:
Heh, that's two, one more of these and it needs to get wrapped up in a helper :)